git-sync/main.go

```diff
@@ -98,10 +101,14 @@ func main() {
 		}
 	}
 
+	if *flAtomic == true && *flSubDir == "" {
+		log.Fatal("error: GIT_SYNC_SUBDIR is required for atomic initial pull")
+	}
+
 	initialSync := true
 	failCount := 0
 	for {
-		if err := syncRepo(*flRepo, *flDest, *flBranch, *flRev, *flDepth); err != nil {
+		if err := syncRepo(*flRepo, *flDest, *flBranch, *flRev, *flSubDir, *flDepth, *flAtomic); err != nil {
```

Comment:
this is looking pretty close to wanting to be pulled out into a struct with methods.